isis/tests/FunctionalTestsCsminit.cpp

```diff
@@ -0,0 +1,218 @@
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/csminit.xml").expanded();
+
+class CSMPluginFixture : public TempTestingFiles {
+  protected:
+      TestCsmPlugin *plugin;
+      Cube *testCube;
+      Pvl label;
+      QString isdPath;
+      QString filename;
+
+      void SetUp() override {
+        TempTestingFiles::SetUp();
+
+        // Create and populate default test ISD
+        json isd;
+        isd["name"] = "test_isd";
+        isd["test_param_one"] = "value_one";
+        isd["test_param_two"] = "value_two";
+
+        isdPath = tempDir.path() + "/default.json";
+        std::ofstream file(isdPath.toStdString());
+        file << isd;
+        file.flush();
+
+        std::ifstream cubeLabel("data/threeImageNetwork/cube1.pvl");
+        cubeLabel >> label;
+        testCube = new Cube();
+        filename = tempDir.path() + "/csminitCube.cub";
+        testCube->fromLabel(filename, label, "rw");
+        testCube->close();
+
+        plugin = new TestCsmPlugin();
+        csm::PluginList pluginList = plugin->getList();
+        bool alreadyFound = false;
+        for (auto const& loadedPlugin : pluginList) {
+          if (loadedPlugin) {
+            if ( (loadedPlugin->getPluginName() != "TestCsmPlugin") || alreadyFound) {
+              // Note: documentation says explicitly not to do this, but I'm doing it
+              // anyway for testing. See pg. 39 API documentation version 3.0.3
+              plugin->removePlugin(loadedPlugin->getPluginName());
+            }
+            else {
+              // Sometimes end up with mulitple copies of the same plugin. If so, remove it so 
+              // it doesn't get considered "2 plugins" by csminit, which causes it to fail.
+              alreadyFound = true;
+            }
+          }
+        }
+      }
+
+      void TearDown() override {
+        if (plugin) {
+          plugin->removePlugin(plugin->getPluginName());
+          delete plugin;
+          plugin = NULL;
+        }
+        if (testCube) {
+          if (testCube->isOpen()) {
+            testCube->close();
+          }
+          delete testCube;
+          testCube = NULL;
+        }
+      }
+};
+
+TEST_F(CSMPluginFixture, CSMInitDefault) {
+  // Run csminit with defaults for everything besides FROM and ISD
+  QVector<QString> args = {
+    "from="+filename,
+    "isd="+isdPath
+  };
+
+  UserInterface options(APP_XML, args);
+  csminit(options);
+
+  testCube->open(filename);
+
+  // Get a model and a state string
+  StringBlob stateString("","CSMState");
+  testCube->read(stateString);
+
+  // Verify contents of the StringBlob's PVL label
+  PvlObject blobPvl = stateString.Label(); 
+  EXPECT_EQ(stateString.Name().toStdString(), "CSMState");
+  EXPECT_EQ(stateString.Type().toStdString(), "String"); 
+
+  // Intent is just to make sure the state string exists and is a "reasonable" length
+  EXPECT_GT(stateString.string().length(), 20);
```

Comment:
Thoughts on a better way to test this? I think this probably isn't the best solution.

Reply:
Check that the plugin can construct a model from the state string.